lib/rules-generator.js

```diff
@@ -123,10 +131,12 @@ util.methods(Generator, {
   },
 
   validateMethods: function(m, methods, allowed) {
-    util.forNotIn(Object.keys(methods), allowed, function(method) {
-      this.log.warn(m + util.quoteString(method) +
-                    " (allowed: " + allowed.map(util.quoteString).join(', ') + ")");
-    }.bind(this));
+    for (var method in methods) {
```

Comment:
I think this is considered dodgy practice. for in loops are reserved for dictionary/objects
In ES6 it's for (method of methods). 
For existing ES5 it's methods.forEach(function(method){
});
For below ES5 e.g. IE 8, it's a crappy counting loop. I dunno which we are supporting. I would go for forEach for arrays. But none of the best practices suggest for (. in ...)